### PR TITLE
enable redfish by default

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

Enable Redfish power management by default

## GUI diff

No difference.

- [x] **DONE**

## Documentation

- Documentation issue was created: https://github.com/SUSE/spacewalk/issues/12767

- [x] **DONE**

## Test coverage
- Cucumber tests were added

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/9866

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: testsuite/features/step_definitions/command_steps.rb [hunk 1]
```diff
@@ -445,6 +445,25 @@
...
 
+When(/^the server starts mocking a Redfish host$/) do
+  $server.run("mkdir -p /root/Redfish-Mockup-Server/")
+  ["redfishMockupServer.py", "rfSsdpServer.py"].each do |file|
+    source = File.dirname(__FILE__) + '/../upload_files/Redfish-Mockup-Server/' + file
+    dest = "/root/Redfish-Mockup-Server/" + file
+    return_code = file_inject($server, source, dest)
+    raise 'File injection failed' unless return_code.zero?
+  end
+  $server.run("curl --output DSP2043_2019.1.zip https://www.dmtf.org/sites/default/files/standards/documents/DSP2043_2019.1.zip")
```
[the remark this answers is not in the document]
Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:456]: Could we please add it to our mirrors? better not have dependencies with external network.

File: testsuite/features/step_definitions/command_steps.rb [hunk 2]
```diff
@@ -445,6 +445,25 @@
   $server.run("kill $(pidof -x fake_ipmi_host.sh)")
 end
 
+When(/^the server starts mocking a Redfish host$/) do
+  $server.run("mkdir -p /root/Redfish-Mockup-Server/")
+  ["redfishMockupServer.py", "rfSsdpServer.py"].each do |file|
```
Comment (srbarrios) [inline testsuite/features/step_definitions/command_steps.rb:450]: I prefer single quotes if we don't have something that needs to be escaped.

File: testsuite/features/step_definitions/command_steps.rb [hunk 3]
```diff
@@ -474,7 +493,13 @@
   node.run('rm /tmp/test_user_defined_state')
 end
 
-Then(/^the cobbler report contains "([^"]*)" for system "([^"]*)"$/) do |arg1, system|
+Then(/^the cobbler report contains "([^"]*)" for system "([^"]*)"$/) do |arg1, host|
+  node = get_target(host)
+  output = sshcmd("cobbler system report --name #{node.full_hostname}:1", ignore_err: true)[:stdout]
+  raise "Not found: #{output}" unless output.include?(arg1)
```
Comment (srbarrios) [inline testsuite/features/step_definitions/command_steps.rb:499]:
```suggestion
  raise "Not found:\n#{output}" unless output.include?(arg1)
```

File: testsuite/features/step_definitions/command_steps.rb [hunk 4]
```diff
@@ -474,7 +493,13 @@
   node.run('rm /tmp/test_user_defined_state')
 end
 
-Then(/^the cobbler report contains "([^"]*)" for system "([^"]*)"$/) do |arg1, system|
+Then(/^the cobbler report contains "([^"]*)" for system "([^"]*)"$/) do |arg1, host|
```
Comment (srbarrios) [inline testsuite/features/step_definitions/command_steps.rb:496]:
```suggestion
Then(/^the cobbler report contains "([^"]*)" for system "([^"]*)"$/) do |text, system|
```

File: testsuite/features/step_definitions/navigation_steps.rb [hunk 5]
```diff
@@ -950,3 +950,8 @@
 When(/^I select the next maintenance window$/) do
   find(:xpath, "//select[@id='maintenance-window-select']/option", match: :first).select_option
 end
+
+When(/^I enter "([^"]*)" as the redfish server address$/) do |host|
```
Comment (srbarrios) [inline testsuite/features/step_definitions/navigation_steps.rb:954]:
```suggestion
When(/^I enter the server hostname as the redfish server address$/) do
```

Comment (srbarrios) [reply testsuite/features/step_definitions/navigation_steps.rb:954]: In my opinion, if we are already sticking into use it only on the server node, in other steps like the stop.
We shouldn't allow here to pass other values. Just create a sentence without parameters, it's fine.

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 6]
```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2020 SUSE LLC
+# Licensed under the terms of the MIT license.
+
+Feature: Power management
+
+  Scenario: Fake a Redfish host
```
Comment (srbarrios) [inline testsuite/features/secondary/srv_power_management_redfish.feature:6]: I would just hide the point that we are moking it, in our Gherkin scenarios.
The implementation we make behind the scenarios shouldn't be important at the test description level

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    Given I am on the Systems overview page of this "sle_minion"
+    And I follow "Provisioning" in the content area
+    And I follow "Power Management" in the content area
+    Then I click on "Power On"
+    And I click on "Get status"
+    And I should see the power is "On"
+    And I click on "Power Off"
+    And I should see a "system has been powered off" text
+    And I wait for "20" seconds
+    And I should see the power is "Unknown"
```
Comment (srbarrios) [inline testsuite/features/secondary/srv_power_management_redfish.feature:34]: Please just add a timeout of 20 seconds inside the step definition of `I should see the power is "Unknown"`, so we can get rid of steps like `I wait for x seconds`

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 8]
```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2020 SUSE LLC
+# Licensed under the terms of the MIT license.
+
+Feature: Power management
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+  Scenario: Fake a Redfish host
+    Given the server starts mocking a Redfish host
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+  Scenario: Save power management values for Redfish
+    Given I am on the Systems overview page of this "sle_minion"
+    And I follow "Provisioning" in the content area
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    And I follow "Power Management" in the content area
+    When I enter "server" as the redfish server address
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+    And I enter "ipmiusr" as "powerUsername"
+    And I enter "test" as "powerPassword"
+    And I select "Redfish" from "powerType"
+    And I click on "Save"
+    Then I should see a "Power settings saved" text
+    And the cobbler report contains ":8443" for system "sle_minion"
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+    And the cobbler report contains "Power Management Username      : ipmiusr" for system "sle_minion"
+    And the cobbler report contains "Power Management Password      : test" for system "sle_minion"
+    And the cobbler report contains "Power Management Type          : redfish" for system "sle_minion"
+
+  Scenario: Test Redfish functions
+    Given I am on the Systems overview page of this "sle_minion"
+    And I follow "Provisioning" in the content area
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+    And I follow "Power Management" in the content area
+    Then I click on "Power On"
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature (hunk 7 above)
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+    And I click on "Get status"
+    And I should see the power is "On"
+    And I click on "Power Off"
+    And I should see a "system has been powered off" text
+    And I wait for "20" seconds
+    And I should see the power is "Unknown"
+    Then I click on "Get status"
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+    And I should see the power is "Off"
+    Then I click on "Power On"
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+    And I follow "Power Management" in the content area
+    Then I click on "Power On"
+    And I click on "Get status"
+    And I should see the power is "On"
+    And I click on "Power Off"
+    And I should see a "system has been powered off" text
+    And I wait for "20" seconds
+    And I should see the power is "Unknown"
+    Then I click on "Get status"
+    And I should see the power is "Off"
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+    And I should see the power is "On"
+    Then I click on "Reboot"
+    And I should see a "system has been rebooted" text
+    Then I click on "Get status"
+    And I should see the power is "On"
+
+  Scenario: Check power management SSM configuration for Redfish
+    Given I am authorized
+    And I am on the System Overview page
+    And I check the "sle_minion" client
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+    Then I should see a "Configuration successfully saved for 1 system(s)" text
+    And the cobbler report contains "Power Management Username      : testing" for system "sle_minion"
+    And the cobbler report contains "Power Management Password      : qwertz" for system "sle_minion"
+    And the cobbler report contains ":8443" for system "sle_minion"
+    And the cobbler report contains "Power Management Type          : redfish" for system "sle_minion"
+
+  Scenario: Check power management SSM operation for Redfish
+    Given I am authorized
+    And I am on System Set Manager Overview
+    And I follow "power management operations" in the content area
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+    Then I should see "sle_minion" as link
+    And I should see a "Power On" button
+    And I should see a "Power Off" button
+    And I should see a "Reboot" button
+
+  Scenario: Cleanup: reset Redfish values
+    Given I am logged in via XML-RPC powermgmt as user "admin" and password "admin"
+    And I want to operate on this "sle_minion"
+    Then I set power management value "" for "powerAddress"
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+    And I set power management value "" for "powerUsername"
+    And I set power management value "" for "powerPassword"
+    And I set power management value "ipmitool" for "powerType"
+    And the cobbler report contains "Power Management Address       :" for system "sle_minion"
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/secondary/srv_power_management_redfish.feature [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+    And the cobbler report contains "Power Management Username      :" for system "sle_minion"
+    And the cobbler report contains "Power Management Password      :" for system "sle_minion"
+    And the cobbler report contains "Power Management Type          : ipmitool" for system "sle_minion"
+
+  Scenario: Cleanup: don't fake a Redfish host
+    Given the server stops mocking a Redfish host
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/step_definitions/command_steps.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/step_definitions/command_steps.rb:456]: Nuremberg instances do not use a mirror. We need to think about a better solution

File: testsuite/features/step_definitions/command_steps.rb [hunk 23]
```diff
@@ -445,6 +445,25 @@
...
+  $server.run("mkdir -p /root/Redfish-Mockup-Server/")
+  ["redfishMockupServer.py", "rfSsdpServer.py"].each do |file|
+    source = File.dirname(__FILE__) + '/../upload_files/Redfish-Mockup-Server/' + file
+    dest = "/root/Redfish-Mockup-Server/" + file
+    return_code = file_inject($server, source, dest)
+    raise 'File injection failed' unless return_code.zero?
+  end
+  $server.run("curl --output DSP2043_2019.1.zip https://www.dmtf.org/sites/default/files/standards/documents/DSP2043_2019.1.zip")
+  $server.run("unzip DSP2043_2019.1.zip")
+  cmd = "/usr/bin/python3 /root/Redfish-Mockup-Server/redfishMockupServer.py -H #{$server.full_hostname} -p 8443 -S " \
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/step_definitions/command_steps.rb]: done, but I tried to keep belonging params together

File: testsuite/features/step_definitions/command_steps.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/step_definitions/command_steps.rb]: done

File: testsuite/features/step_definitions/command_steps.rb [hunk 24]
```diff
@@ -474,7 +493,13 @@
...
 end
 
-Then(/^the cobbler report contains "([^"]*)" for system "([^"]*)"$/) do |arg1, system|
+Then(/^the cobbler report contains "([^"]*)" for system "([^"]*)"$/) do |arg1, host|
+  node = get_target(host)
+  output = sshcmd("cobbler system report --name #{node.full_hostname}:1", ignore_err: true)[:stdout]
+  raise "Not found: #{output}" unless output.include?(arg1)
+end
+
+Then(/^the cobbler report contains "([^"]*)" for cobbler system "([^"]*)"$/) do |arg1, system|
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/step_definitions/command_steps.rb]: done

File: testsuite/features/step_definitions/command_steps.rb (hunk 2 above)
Comment (mcalmer) [reply testsuite/features/step_definitions/command_steps.rb]: done

File: testsuite/features/step_definitions/command_steps.rb (hunk 3 above)
Comment (mcalmer) [reply testsuite/features/step_definitions/command_steps.rb]: done

File: testsuite/features/step_definitions/command_steps.rb (hunk 4 above)
Comment (mcalmer) [reply testsuite/features/step_definitions/command_steps.rb]: not done. We use "host" when we expect something like 'sle_client' or 'sle_minion'.

File: testsuite/features/step_definitions/navigation_steps.rb (hunk 5 above)
Comment (mcalmer) [reply testsuite/features/step_definitions/navigation_steps.rb]: done

File: testsuite/features/step_definitions/command_steps.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:456]: Ok, then we might need something like that:

```
def use_mirror(domain)
  if ENV['SERVER'].include? 'mgr.prv.suse.net'
    domain
  else
    'minima-mirror.mgr.prv.suse.net'
  end
end
```

```
 $server.run("curl --output DSP2043_2019.1.zip https://#{use_mirror('www.dmtf.org')}/sites/default/files/standards/documents/DSP2043_2019.1.zip")

```

[the remark this answers is not in the document]
Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:456]: Anyhow, nevermind, we can make it in a different PR, don't bother with that for now...

File: testsuite/features/secondary/srv_power_management_redfish.feature (hunk 7 above)
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: I think the wait is useless. Maybe a artifact from the times when we used a real IPMI system. I will remove it

File: testsuite/features/secondary/srv_power_management_redfish.feature (hunk 6 above)
Comment (mcalmer) [reply testsuite/features/secondary/srv_power_management_redfish.feature]: done

File: testsuite/features/step_definitions/command_steps.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/step_definitions/command_steps.rb:456]: I thought to put it on one of our server, but for Uyuni users this is not reachable. So downloading from DMTF directly seems to be the best approach for now. Feel free to come up with a solution which works for us and the community.

[the remark this answers is not in the document]
Comment (mcalmer) [reply testsuite/features/step_definitions/command_steps.rb:456]: Yes, let's think twice and do it in a different PR

**Review by srbarrios — APPROVED** [review]

**Review by Bischoff — APPROVED** [review]
Thanks a lot for the complete fixing of When / Then / And

**Pull request merged**